src/lib/comms/sol-lwm2m-server.c

```diff
@@ -1295,26 +1295,41 @@ send_management_packet(struct sol_lwm2m_server *server,
     return -ENOMEM;
 }
 
-/**
- * This checks if the path has the following form: /2/0/1
- */
-static bool
-is_resource_set(const char *path)
+enum lwm2m_path_props {
+    PATH_IS_INVALID = (1 << 0),
+    PATH_HAS_OBJECT =  (1 << 1),
+    PATH_HAS_INSTANCE = (1 << 2),
+    PATH_HAS_RESOURCE = (1 << 3)
+};
+
+static enum lwm2m_path_props
+get_path_props(const char *path)
 {
-    size_t i;
-    uint8_t slashes;
-    const char *last_slash;
+    size_t i, slashes;
+    enum lwm2m_path_props props = PATH_IS_INVALID;
 
     for (i = 0, slashes = 0; path[i]; i++) {
         if (path[i] == '/') {
-            last_slash = path + i;
+            props =  props << 1;
             slashes++;
+            if (slashes > 3) {
+                SOL_WRN("The path '%s' has an invalid format."
+                    " Expected: /Object/Instance/Resource", path);
+                return PATH_IS_INVALID;
+            }
+        } else if (!isdigit(path[i])) {
+            SOL_WRN("The path '%s' contains a nondigit character: '%c'",
+                path, path[i]);
+            return PATH_IS_INVALID;
         }
     }
 
-    if (slashes < 3 || *(last_slash + 1) == '\0')
-        return false;
-    return true;
+    if (i - slashes == 0) {
```

Comment:
Since this means "The `path` is only formed by `slashes` and has no digits", wouldn't it be easier to read an `if (i == slashes)`?